driver-sync/src/main/com/mongodb/client/internal/KeyManagementService.java

```diff
@@ -46,7 +48,7 @@ public InputStream stream(final String host, final ByteBuffer message) throws IO
         try {
             enableHostNameVerification(socket);
             socket.setSoTimeout(timeoutMillis);
-            socket.connect(serverAddress.getSocketAddress(), timeoutMillis);
+            socket.connect(new InetSocketAddress(InetAddress.getByName(serverAddress.getHost()), serverAddress.getPort()), timeoutMillis);
```

Comment:
Stop using ServerAddress method because it wraps the UnknownHostException in a MongoSocketException, and that makes the exception handling inconsistent between sync and reactive.